src/nupic/algorithms/Connections.cpp

```diff
@@ -557,6 +557,44 @@ void Connections::bumpSegment(const Segment segment, const Permanence delta) {
 }
 
 
+void Connections::destroyMinPermanenceSynapses(Segment segment, Int nDestroy,
```

Comment:
I'd like to keep this as signed, and then assert that it's not less than 0.  This way it detects underflow errors instead of hiding them.

Reply:
> This way it detects underflow errors instead of hiding them.
 
ok. The calling code still needs be fixed. see size_t overrun in TM.cpp, which underflows :grin:

Reply:
I think David Keeney fixed this on master branch, thanks!